app/controllers/api/v1/accounts/pins_controller.rb

```diff
@@ -8,7 +8,9 @@ class Api::V1::Accounts::PinsController < Api::BaseController
   before_action :set_account
 
   def create
-    AccountPin.create!(account: current_account, target_account: @account)
+    unless AccountPin.exists?(account: current_account, target_account: @account)
+      AccountPin.create!(account: current_account, target_account: @account)
+    end
```

Comment:
`AccountPin.find_or_create_by!`? It's functionally equivalent (that is, still not doing the change atomically) but more idiomatic